packages/ui/src/composables/useBem.ts

```diff
@@ -1,21 +1,59 @@
-import { computed, Ref, unref } from 'vue'
+import { computed, Ref, unref, ComputedRef } from 'vue'
+import { __DEV__ } from '../utils/global-utils'
 
-export const useBem = (baseClass: string) => {
-  const createModifiersClasses = (modifiersCb: () => Record<string, boolean>, classes: string[] | Ref<string[]> = []) => {
-    return computed(() => {
-      const modifiers = modifiersCb()
-      return [
-        baseClass,
-        ...unref(classes),
-        ...Object
-          .keys(modifiers)
-          .filter((modifierName) => modifiers[modifierName])
-          .map((modifierName) => `${baseClass}--${modifierName}`),
-      ]
-    })
-  }
+type ComputedClasses = Ref<Record<string, any>> & {
+  value: ComputedRef<Record<string, true>>
```

Comment:
Since Ref already have value, why do we need extra value property?

Reply:
Because we want to have this:
![](http://joxi.ru/4AkXWMXCjRodgr.jpg)